src/azure-cli/azure/cli/command_modules/privatedns/custom.py

```diff
@@ -24,6 +27,231 @@ def list_privatedns_zones(cmd, resource_group_name=None):
+
+            for entry in record_set_obj:
+
+                record_set_ttl = entry['ttl']
+                record_set_key = '{}{}'.format(record_set_name.lower(), record_set_type)
+
+                record = _build_record(cmd, entry)
+                if not record:
+                    logger.warning('Cannot import %s. RecordType is not found. Skipping...', entry['delim'].lower())
+                    continue
+
+                record_set = record_sets.get(record_set_key, None)
+                if not record_set:
+
+                    # Workaround for issue #2824
+                    relative_record_set_name = record_set_name.rstrip('.')
+                    if not relative_record_set_name.endswith(origin):
+                        logger.warning(
+                            'Cannot import %s. Only records relative to origin may be '
+                            'imported at this time. Skipping...', relative_record_set_name)
+                        continue
+
+                    record_set = RecordSet(ttl=record_set_ttl)
+                    record_sets[record_set_key] = record_set
+                _privatedns_add_record(record_set, record, record_set_type, is_list=record_set_type.lower() not in ['soa', 'cname'])
+
+    total_records = 0
+    for key, rs in record_sets.items():
+        rs_name, rs_type = key.lower().rsplit('.', 1)
+        rs_name = rs_name[:-(len(origin) + 1)] if rs_name != origin else '@'
+        try:
+            record_count = len(getattr(rs, _privatedns_type_to_property_name(rs_type)))
+        except TypeError:
+            record_count = 1
+        total_records += record_count
+    cum_records = 0
+
+    from azure.mgmt.privatedns import PrivateDnsManagementClient
+    from azure.mgmt.privatedns.models import PrivateZone
+    client = get_mgmt_service_client(cmd.cli_ctx, PrivateDnsManagementClient)
+
+    print('== BEGINNING ZONE IMPORT: {} ==\n'.format(private_zone_name), file=sys.stderr)
+
+    if private_zone_name.endswith(".local"):
+        logger.warning(("Please be aware that DNS names ending with .local are reserved for use with multicast DNS "
+                        "and may not work as expected with some operating systems. For details refer to your operating systems documentation."))
+    zone = PrivateZone(location='global')
+    client.private_zones.create_or_update(resource_group_name, private_zone_name, zone)
+
+    for key, rs in record_sets.items():
+
+        rs_name, rs_type = key.lower().rsplit('.', 1)
+        rs_name = '@' if rs_name == origin else rs_name
+        if rs_name.endswith(origin):
+            rs_name = rs_name[:-(len(origin) + 1)]
+
+        try:
+            record_count = len(getattr(rs, _privatedns_type_to_property_name(rs_type)))
+        except TypeError:
+            record_count = 1
+        if rs_name == '@' and rs_type == 'soa':
+            # 30 second sleep in case zone is getting created. else follwoing get soa record-set call fails
+            time.sleep(30)
```

Comment:
Instead of a 30 sec sleep, you should poll for the private zone creation operation and proceed only if it succeeds.

Reply:
Please use `LongRunningOperation`

Reply:
Incorporated review comment.